packages/gatsby/src/redux/reducers/schema.ts

```diff
@@ -0,0 +1,13 @@
+import { ActionsUnion, IGatsbyState } from "../types"
+
+export const schemaReducer = (
+  state: IGatsbyState["schema"] | {} = {},
```

Comment:
This isn't quite what I was getting at. I think it would better if we could default this state to an instance of graphql schema. There are APIs and keys that are expected to be on this state. It's fragile to not have them there, even though we must be supporting that as is right now.